kpi/views/v2/data.py

```diff
@@ -531,16 +530,11 @@ def retrieve(self, request, pk, *args, **kwargs):
         # Join all parameters to be passed to `deployment.get_submissions()`
         params.update(filters)
 
-        # The `get_submissions()` is a generator in KobocatDeploymentBackend
-        # class but a list in MockDeploymentBackend. We cast the result as a list
-        # no matter what is the deployment back-end class to make it work with
-        # both. Since the number of submissions is very small, it should not
-        # have a big impact on memory (i.e. list vs generator)
-        submissions = list(deployment.get_submissions(**params))
+        submissions = deployment.get_submissions(**params)
         if not submissions:
             raise Http404
 
-        submission = submissions[0]
+        submission = list(submissions)[0]
```

Comment:
You are already returning a list in https://github.com/kobotoolbox/kpi/blob/d7bd068b1b2d050d9cbec147d805fd8ca50f9bcd/kpi/deployment_backends/mock_backend.py#L45-L49

Reply:
Indeed, but OpenRosaBackend is returning a generator.

Reply:
Ha, not sure why I thought this was test code :facepalm: Thanks